crates/store/src/db/mod.rs

```diff
@@ -39,29 +43,38 @@ pub struct NullifierInfo {
 }
 
 #[derive(Debug, Clone, PartialEq)]
-pub struct NoteCreated {
-    pub batch_index: u32,
-    pub note_index: u32,
+pub struct Note {
+    pub block_num: BlockNumber,
+    pub note_index: BlockNoteIndex,
     pub note_id: RpoDigest,
     pub note_type: NoteType,
     pub sender: AccountId,
     pub tag: u32,
     pub details: Option<Vec<u8>>,
+    pub merkle_path: MerklePath,
 }
 
-impl NoteCreated {
+impl Note {
     /// Returns the absolute position on the note tree based on the batch index
     /// and local-to-the-subtree index.
     pub fn absolute_note_index(&self) -> u32 {
-        BlockNoteTree::note_index(self.batch_index as usize, self.note_index as usize) as u32
+        self.note_index.note_index() as u32
     }
 }
```

Comment:
nit: I'm wondering if we actually need this `absolute_note_index()`. Seems like we can just call `note.note_index.note_index()` (i think it would have been better if it was named not `note_index()` but something like `to_absolute_index()`.

Reply:
Okay, I renamed function in that PR: https://github.com/0xPolygonMiden/miden-base/pull/630